### PR TITLE
Models creation

### PR DESCRIPTION
**Pull request by redmarverick**
### [Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md):

- Create models:

1. Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png). ✔️
2. Set up associations between models. ✔️
3. Remember that author_id column in posts table should be the foreign_key for the users table. ✔️

- Use models to insert data:

1. Create at least one user ✔️
2. Create at least 4 posts written by one of the users you created ✔️
3. Create at least 6 posts comments for one of the posts you created ✔️
4. Use other [CRUD methods] ✔️

- Create custom methods:

1. A method that returns the 3 most recent posts for a given user. ✔️
2. A method that updates the posts counter for a user. ✔️
3. A method which returns the 5 most recent comments for a given post. ✔️
4. A method that updates the comments counter for a post. ✔️
5. A method that updates the likes counter for a post. ✔️
6. Go to rails c and check if your methods are working.✔️

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
```
Comment (sumairq) [inline app/models/user.rb:2]: - [x] A user model needs to have associations for likes and comments as well. Please add them like this 
 ```
 has_many :comments, foreign_key: 'author_id'
 has_many :likes, foreign_key: 'author_id'
 ```

**Review by sumairq — CHANGES_REQUESTED** [review]
### Hi  @redmarverick , 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Professional README
✔️Correct Git flow
✔️Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (redmarverick) [reply app/models/user.rb:2]: correction done

**Review by aamirkhan2478 — APPROVED** [review]
Hi @redmarverick, 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**